src/main/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProvider.java

```diff
@@ -28,6 +28,10 @@ public abstract class DisplayURLProvider implements ExtensionPoint {
      * @return DisplayURLProvider
      */
     public static DisplayURLProvider get() {
+        // TODO: Would it not make more sense to return DisplayURLProviderImpl.INSTANCE unconditionally, always
+        // serving /display/redirect URLs that only get resolved when an actual user clicks on them? There is no
+        // guarantee that the current user (if any) is going to be the user who clicks the generated link so IDK why we
+        // take the current user's preferences into account here.
```

Comment:
This method only gets used when _generating_ links, not when following them, so it doesn't make sense to me to consider user preference here. In practice I don't think it matters because almost no one uses `authorize-project` and so this always gets called in a context where there is no active user, but it is confusing.

Reply:
Turns out it used to work the way I had in mind prior to #42, and [JENKINS-69006](https://issues.jenkins.io/browse/JENKINS-69006) is asking for it to be switched back, so I will look into fixing that issue while I am here.